### PR TITLE
Update all solution projects to .NET 6

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

File: perf/benchmarkapps/GrpcAspNetCoreServer/GrpcAspNetCoreServer.csproj [hunk 1]
```diff
@@ -40,7 +40,7 @@
     <ProjectReference Include="..\..\..\src\Grpc.AspNetCore.Web\Grpc.AspNetCore.Web.csproj" />
   </ItemGroup>
   
-  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0' OR '$(TargetFramework)' == 'net6.0'">
+  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0' OR '$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net7.0'">
```
Comment (captainsafia) [inline perf/benchmarkapps/GrpcAspNetCoreServer/GrpcAspNetCoreServer.csproj:43]:
```suggestion
  <ItemGroup Condition="$([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">
```

To spare our future selves!

File: perf/benchmarkapps/GrpcClient/GrpcClient.csproj [hunk 2]
```diff
@@ -42,7 +42,7 @@
     <ProjectReference Include="..\..\..\src\Grpc.AspNetCore.Web\Grpc.AspNetCore.Web.csproj" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0' OR '$(TargetFramework)' == 'net6.0'">
+  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0' OR '$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net7.0'">
```
Comment (captainsafia) [inline perf/benchmarkapps/GrpcClient/GrpcClient.csproj:45]:
```suggestion
  <ItemGroup Condition="$([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">
```

File: testassets/InteropTestsClient/InteropTestsClient.csproj [hunk 3]
```diff
@@ -43,7 +43,7 @@
     <PackageReference Include="System.Net.Http.WinHttpHandler" Version="$(SystemNetHttpWinHttpHandlerPackageVersion)" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)'=='net5.0'">
+  <ItemGroup Condition="'$(TargetFramework)'=='net5.0' OR '$(TargetFramework)'=='net6.0'">
```
Comment (captainsafia) [inline testassets/InteropTestsClient/InteropTestsClient.csproj:46]:
```suggestion
  <ItemGroup Condition="$([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">
```

File: testassets/InteropTestsWebsite/InteropTestsWebsite.csproj [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFrameworks Condition="'$(LatestFramework)'!='true'">net6.0</TargetFrameworks>
-    <TargetFrameworks Condition="'$(LatestFramework)'=='true'">net5.0</TargetFrameworks>
+    <TargetFrameworks Condition="'$(LatestFramework)'=='true'">net6.0</TargetFrameworks>
```
Comment (captainsafia) [inline testassets/InteropTestsWebsite/InteropTestsWebsite.csproj:5]: Is there any point in having these two conditions at all if we are defaulting to `net6.0` anyways?

**Review by captainsafia — COMMENTED** [review]
Sensible change overall! One question inline.

**Review by captainsafia — APPROVED** [review]

File: perf/benchmarkapps/GrpcAspNetCoreServer/GrpcAspNetCoreServer.csproj (hunk 1 above)
Comment (JamesNK) [reply perf/benchmarkapps/GrpcAspNetCoreServer/GrpcAspNetCoreServer.csproj:43]: TIL!

File: testassets/InteropTestsWebsite/InteropTestsWebsite.csproj (hunk 4 above)
Comment (JamesNK) [reply testassets/InteropTestsWebsite/InteropTestsWebsite.csproj:5]: I left it as is because it will change when the SDK and projects update to net7.

**Pull request merged**